### ISSUE TITLE
Customize Handles

### ISSUE THREAD

**Issue by Gibbo3771** [aeagle/react-spaces#97]
**Is your feature request related to a problem? Please describe.**
Handles can't be customized

**Describe the solution you'd like**
I want to customised the handles so that a custom component is rendered

**Describe alternatives you've considered**
I can't see anything except forking and changing it under the hood

Is there something I am missing here? I am currently exploring library options for a large re-write of a commercial application and this is one of the cleanest and well documented solutions that does exactly what we need.

**Comment by aeagle** [comment]
Hi @Gibbo3771. There is a similar request on #54. Out of interest, would you say the custom component you want to use for a resize handle has any element other than CSS styling? Potentially would you be able to set the resize handle size (with the `handleSize` prop) and then style the .spaces-resize-handle element on the page?

**Comment by Gibbo3771** [comment]
I actually made a PR with what I had in mind #98.

The change would give user complete control over how the handle is rendered, including it's props. The prop would take a function that is called for every handle on the space. This function will have access to the props for that particular handle.

```jsx
<LeftResizable handleRender={(props) => {
        return <div {...props}></div>
    }
>
    // content
</LeftResizable>
```

The above example would be the same as if you never used the `handleRender` prop, but you can see the possibilities since they the developer will have access to the same data the underlying API does.

**Comment by aeagle** [comment]
@Gibbo3771 That's useful and I will definitely consider it, but just so I consider all alternatives is it because you want to provide your own styled handle or is there something else you would be doing there "with full access" to the resize API?

**Comment by Gibbo3771** [comment]
This mostly came from me searching for a framework for a re-write of a large commercial app, we have specific styling and dragging anchor points on our current implementation, that we want to keep. Yes I could override the CSS, but I felt exposing the API in a controlled manner would be more appropriate.

Everyone has different needs, and I guess it comes down to _does this fit with your vision_.

**Comment by aeagle** [comment]
Yes absolutely. I just want to check that I am not boxing myself in with
stuff I have in progress at the moment. I've got a lots of work on a
resize-handles branch for positional spaces which are not anchored (think
resizable windows kind of thing). Let me check compatibility with what's
there.

**Comment by aeagle** [comment]
Prepping function in #103

**Comment by aeagle** [comment]
Functionality has now been released in `v0.3.0`
